flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobVertex.java

```diff
@@ -125,6 +132,8 @@ public JobVertex(String name) {
 	public JobVertex(String name, JobVertexID id) {
 		this.name = name == null ? DEFAULT_NAME : name;
 		this.id = id == null ? new JobVertexID() : id;
+		this.operatorIDs.add(new OperatorID(this.id.getLowerPart(), this.id.getUpperPart()));
+		this.operatorIdsAlternatives.add(null);
```

Comment:
Ok, in this cases it really seems better to make this explicit as you suggested. Also I was wondering if `operatorIdsAlternatives` should be a List<List<OperatorID>> -- just want to make sure that only at most one alternative ID must be maintained per operator. But I think that we can always determine the savepoint version and only need compatibility to the hasher version that was valid under that savepoint version.